addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,194 @@
+                    return msg;
+                } catch (Exception e) {
+                    // Check if it is a connection error
+                    if (e.getMessage().contains("ConnectionError")
+                            || e.getMessage().contains("IOException")) {
+                        continue;
+                    }
+                    logger.info(e);
+                    errorCount = errorCount + 1;
+                }
+            }
+            logger.info("Total number of Errors: " + errorCount);
+        } catch (URIException e) {
+            logger.error("Error parsing URL: {} Exception raised {}", url, e);
+        }
+        return null;
+    }
+
+    /**
+     * Returns a Map of all parameters to be used from a given wordlist along with a fuzzy number
+     * key.
+     *
+     * @param list the List of all parameters to be used from a given wordlist.
+     * @return a Map<String, String> of all parameters to be used from a given wordlist along with a
+     *     fuzzy number key.
+     */
+    public static Map<String, String> populate(List<String> list) {
+        Map<String, String> map = new HashMap<>();
+        String key;
+        String s;
+        for (int i = 0; i < list.size(); i++) {
+            s = list.get(i);
+            key = StringUtils.repeat("1", (6 - Integer.toString(i).length())) + Integer.toString(i);
+            map.put(s, key);
+        }
+        logger.info("Populated map with " + map.size() + " keys!");
+        return map;
+    }
+
+    /**
+     * Divides a given Map of parameters into given parts.
+     *
+     * @param parts the number of parts to divide the Map into.
+     * @param map the Map of parameters to be divided.
+     * @return a List<Map<String, String>> of parameters divided into given parts.
+     */
+    public static List<Map<String, String>> slice(Map<String, String> map, int parts) {
+        List<Map<String, String>> list = new ArrayList<>();
+        int size = map.size();
+        List<Entry<String, String>> items = new ArrayList<>(map.entrySet());
+        int k = size / parts;
+        int m = size % parts;
+        for (int i = 0; i < parts; i++) {
+            int x = i * k + Math.min(i, m);
+            int y = (i + 1) * k + Math.min(i + 1, m);
+            Map<String, String> newMap = new HashMap<>();
+            for (int j = x; j < y; j++) {
+                newMap.put(items.get(j).getKey(), items.get(j).getValue());
+            }
+            list.add(newMap);
+        }
+        return list;
+    }
+
+    /**
+     * Returns a query string from a given parameter map.
+     *
+     * @param map the Map of parameters to be used.
+     * @return a query string from a given parameter map.
+     */
+    public static String createQueryString(Map<String, String> map) {
+        String queryString = "";
+        for (Entry<String, String> entry : map.entrySet()) {
+            queryString += entry.getKey() + "=" + entry.getValue() + "&";
```

Comment:
It's not encoding the names nor the values.

Reply:
It shouldn't really encode it.

Reply:
Mention that in the JavaDoc.

Reply:
Okay